### ISSUE TITLE
Real-ESRGAN slow ONNX inference compared to Pytorch one

### ISSUE THREAD

**Issue by rogachevai** [onnx/onnx#4232]
I converted available model from https://github.com/xinntao/Real-ESRGAN, its architecture can be found here: https://github.com/XPixelGroup/BasicSR/blob/master/basicsr/archs/rrdbnet_arch.py

I did it as follows:
```
import torch
import torch.onnx
from basicsr.archs.rrdbnet_arch import RRDBNet

model = RRDBNet(num_in_ch=3, num_out_ch=3, num_feat=64, num_block=23, num_grow_ch=32, scale=4)

model.load_state_dict(torch.load('../experiments/pretrained_models/RealESRGAN_x4plus.pth')['params_ema'])

model.train(False)
model.cuda().eval()

x = torch.rand(1, 3, 64, 64).cuda()
input_names=['data']
output_names=['output']
dynamic_axes= {'data':{2:'width', 3:'height'}, 'output':{2:'width', 3:'height'}}

with torch.no_grad():
    torch_out = torch.onnx._export(model, x, 'realesrgan-x4-default-cuda-one-batch.onnx', input_names=input_names, output_names=output_names, dynamic_axes=dynamic_axes, opset_version=11, export_params=True)
```

Then I inference it like this:
```
sess = rt.InferenceSession('realesrgan-x4-default-cuda-one-batch.onnx', providers=['CUDAExecutionProvider'])
print("loaded model.")

in_image = cv2.imread('test_img/test.jpg', cv2.IMREAD_UNCHANGED)
print("loaded input image.")

in_mat = cv2.cvtColor(in_image, cv2.COLOR_BGR2RGB)
in_mat = np.transpose(in_mat, (2, 1, 0))[np.newaxis]
in_mat = in_mat.astype(np.float32)
in_mat = in_mat/255

start_time = time.time()
input_name = sess.get_inputs()[0].name
output_name = sess.get_outputs()[0].name
in_mat = torch.tensor(in_mat).cuda()
out_mat = sess.run([output_name], {input_name: in_mat.cpu().numpy()})[0]
elapsed_time = time.time() - start_time
print(elapsed_time)

loaded model.
loaded input image.
3.9808552265167236
```

In case I apply the model to the same input its elapsed time definitely decreases, however in case I replace it with another picture with different shape (and different output shape as in my case output_shape = 4 * input shape) inference time is still bad. 
I tried to do a warmup with a "big" input shape however I noticed no improvements.
Maybe I should somehow update my code in order to avoid moving input to cpu in order to provide `sess` with numpy input? 

So main points here that I need to use batches with size one, each input is torch on cuda device initially and all the inputs and outputs are going to have different shapes. 

I'm using:
Pytorch 1.10.0+cu111
onnx 1.11.0
onnxruntime 1.11.1

**Comment by jcwchen** [comment]
Hi @rogachevai,
This performance issue seems more related to ONNX Runtime. Could you please raise this issue there to get best help from runtime experts? Thanks!
